crates/wasi-provider/src/states/registered.rs

```diff
@@ -0,0 +1,156 @@
+use log::{error, info};
+
+use crate::{make_status, PodState};
+use kubelet::state::{State, Transition};
+use kubelet::{
+    container::Container,
+    pod::{Phase, Pod},
+    state,
+};
+
+use super::error::Error;
+use super::image_pull::ImagePull;
+
+fn validate_pod_runnable(pod: &Pod) -> anyhow::Result<()> {
```

Comment:
These look lifted from the `wascc` provider. Are the necessary for `wasi`?

Reply:
No, they're unnecessary. I left these in as a TODO reminder. The first stage was to get it compiling again.